src/lib/coerce.js

```diff
@@ -99,16 +99,18 @@ exports.valObjects = {
         // TODO 'values shouldn't be in there (edge case: 'dash' in Scatter)
         otherOpts: ['dflt', 'noBlank', 'strict', 'arrayOk', 'values'],
         coerceFunction: function(v, propOut, dflt, opts) {
-            if(opts.strict === true && typeof v !== 'string') {
-                propOut.set(dflt);
-                return;
-            }
+            if(typeof v !== 'string') {
+                var okToCoerce = (
+                    typeof v === 'number' ||
+                    v instanceof Date ||
+                    typeof v === 'boolean'
+                );
```

Comment:
done in https://github.com/plotly/plotly.js/pull/697/commits/ad8db59d9a2a508ddd7ed19d5ba759f58ca7b8e4